src/beman/exemplar/CMakeLists.txt

```diff
@@ -31,31 +34,31 @@ if(BEMAN_EXEMPLAR_INSTALL_CONFIG_FILE_PACKAGE)
     include(CMakePackageConfigHelpers)
 
     configure_package_config_file(
-        "${PROJECT_SOURCE_DIR}/cmake/${PROJECT_NAME}-config.cmake.in"
-        "${PROJECT_BINARY_DIR}/cmake/${PROJECT_NAME}/${PROJECT_NAME}-config.cmake"
-        INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
+        "${CMAKE_CURRENT_SOURCE_DIR}/beman.exemplar-config.cmake.in"
+        "${CMAKE_CURRENT_BINARY_DIR}/beman.exemplar-config.cmake"
+        INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/beman.exemplar"
         PATH_VARS PROJECT_NAME PROJECT_VERSION
     )
 
     write_basic_package_version_file(
-        "${PROJECT_BINARY_DIR}/cmake/${PROJECT_NAME}/${PROJECT_NAME}-version.cmake"
+        "${CMAKE_CURRENT_BINARY_DIR}/beman.exemplar-version.cmake"
         VERSION ${PROJECT_VERSION}
-        COMPATIBILITY SameMajorVersion
+        COMPATIBILITY ExactVersion
     )
 
     install(
         FILES
-            "${PROJECT_BINARY_DIR}/cmake/${PROJECT_NAME}/${PROJECT_NAME}-config.cmake"
-            "${PROJECT_BINARY_DIR}/cmake/${PROJECT_NAME}/${PROJECT_NAME}-version.cmake"
-        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
-        COMPONENT development
+            "${CMAKE_CURRENT_BINARY_DIR}/beman.exemplar-config.cmake"
+            "${CMAKE_CURRENT_BINARY_DIR}/beman.exemplar-version.cmake"
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/beman.exemplar"
+        COMPONENT beman.exemplar
```

Comment:
> Specify an installation component name with which the install rule is associated, such as `Runtime` or `Development`. 
https://cmake.org/cmake/help/latest/command/install.html

Is there a reason to not use a generic name for the component?

Reply:
Discussed off-line. This works around some peculiar cases with composition of cmake projects, and in particular the fragility of PROJECT_NAME and how it is set and propoagated.